gnomad_tgps.py

```diff
@@ -0,0 +1,86 @@
+    for line in p1_file:
+        sid = line.strip()
+        p1_ids.append(f'{sid}_{sid}')
+p1_ids = hl.set(p1_ids)
+
+# Divide the table with genetic data into sample (reference) and out-of-sample populations
+mt1kg_oos = mt1kg.filter_cols(p1_ids.contains(mt1kg.s), keep=False)
+mt1kg_s = mt1kg.filter_cols(p1_ids.contains(mt1kg.s), keep=True)
+
+# annotate the original matrix tables with AC and AF frequencies
+mt1kg_oos = mt1kg_oos.annotate_rows(freq=gnomad_freq_ht[mt1kg_oos.row_key].freq)
+mt1kg_s = mt1kg_s.annotate_rows(freq=gnomad_freq_ht[mt1kg_s.row_key].freq)
+
+# In the out-of-sample (OOS) dataset, filter out SNVs with minor allele frequencies lower than 1%
+# mt1kg_oos = mt1kg_oos.filter_rows(mt1kg_oos.variant_qc.AF[1] > 0.01)
+mt1kg_oos = mt1kg_oos.filter_rows(mt1kg_oos.freq.AF[1] > 0.01)
+
+# In the sample population, filter ouft SNVs with minor allele COUNT lower than 1 -> keep only those SNVs that have a minor allele carrier in the reference group
+# mt1kg_s = mt1kg_s.filter_rows(mt1kg_s.variant_qc.AC[1] > 0)
+mt1kg_s = mt1kg_s.filter_rows(mt1kg_s.freq.AC[1] > 0)
+
+# Annotate the OOS dataset to count the number of minor alleles carried by each participant
+mt1kg_oos = mt1kg_oos.annotate_cols(tn_var_per_sample=hl.agg.count_where(mt1kg_oos.GT.is_non_ref()))
+# Create a new matrix with only SNVs that are not included in the reference group
+mt1kg_oos_filtered = mt1kg_oos.anti_join_rows(mt1kg_s.rows())
+# Annotate the new dataset to count the number of minor alleles carried by each participant
+mt1kg_oos_filtered = mt1kg_oos_filtered.annotate_cols(
+    n_var_per_sample=hl.agg.count_where(mt1kg_oos_filtered.GT.is_non_ref()))
+
+# Keep only the columns (sample meta data)
+mt1kg_oos_filtered_ht = mt1kg_oos_filtered.cols()
+
+# annotate with population data, both continental and subcontinental ancestry
+# genetic region?
+mt1kg_oos_filtered_ht = mt1kg_oos_filtered_ht.annotate(
+    population=table_1kg[mt1kg_oos_filtered_ht.s].population_inference.pop)
+
+ht = mt1kg_oos_filtered_ht
+# annotate with fraction of variants carried by each sample that is not included in the reference
+ht = ht.annotate(fraction=ht.n_var_per_sample / ht.tn_var_per_sample)
+
+# compute aggretation stats
+res = ht.aggregate(hl.struct(ght=hl.agg.group_by(ht.population, hl.agg.mean(ht.n_var_per_sample)),
+                             ghtt=hl.agg.group_by(ht.population, hl.agg.mean(ht.tn_var_per_sample)),
+                             ghtf=hl.agg.group_by(ht.population, hl.agg.mean(ht.fraction)),
+                             ght_stats=hl.agg.group_by(ht.population, hl.agg.stats(ht.n_var_per_sample)),
+                             ghtt_stats=hl.agg.group_by(ht.population, hl.agg.stats(ht.fraction)),
+                             ghtf_stats=hl.agg.group_by(ht.population, hl.agg.stats(ht.tn_var_per_sample))))
+
+# output directory
+tmp_dir = "gs://jialan-tmp-7day/gnomad"
+
+# write to output
+with hfs.open(f'{tmp_dir}/stats.txt', 'w') as f:
+    print(res, file=f)
```

Comment:
we typically use loggers rather than prints (also there is a missing newline here)